### DISCUSSION TITLE
How do I remove a pooling layer and it's subsequent layers from a pretrained network?

### DISCUSSION THREAD

**Discussion by VHK459** [pyg-team/pytorch_geometric#9470] [Q&A] [answered]
I have defined a custom model, which has a graph mean pooling layer. I would like to remove it along with its subsequent layers. Is there a way to do this in pytorch geometric?

**Comment by rusty1s** [comment, marked as answer]
You could use model hooks to get intermediate outputs before the pooling layer. Alternatively, you can use `torch.fx` to modify the model on the fly.
